doc/user_guide/big_data.rst

```diff
@@ -0,0 +1,228 @@
+.. _big-data-label:
+
+Big data
+********
+
+.. versionadded:: 1.2
+
+This chapter describes the behaviour of :py:class:`~._signals.lazy.LazySignal`
+class and its derivatives.
+
+The LazySignal idea
+-------------------
+
+In usual (non-lazy) HyperSpy signals the actual data (accessed via the field
+``s.data``) is most often loaded into the computer memory for fast access and
+processing. While such behaviour indeed results in best performance, it
+obviously requires at least as much computer memory as the data, often twice
+that to store the results of any computation. This becomes a significant
+problem if processing large amounts of data on consumer-oriented hardware.
```

Comment:
"Standard HyperSpy signals load the data (accessed via the field ``s.data``) into memory for fast access and processing. While this behaviour gives good performance in terms of speed, it obviously requires at least as much computer memory as the dataset, and often twice that to store the results of subsequent computation. This can become a significant problem when processing very large datasets on consumer-oriented hardware."